src/System.Runtime.Extensions/tests/project.lock.json

```diff
@@ -1,5 +1,5 @@
 {
-  "locked": true,
+  "locked": false,
```

Comment:
Thanks for pointing out that bug. A few remarks:

1.- I tried it , modifying it so the constructor of HttpClient receives a WinHttpHandler and touches the modified code. I could repro the results without my changes.

2.- After adding my changes the bug is still there, the property IsCancellationRequested doesn't seem to propagate to the cancellation token of the exception being thrown. I'm still investigating why and what part of the stack is actually in charge of throwing that exception. 

3.- I know that my modified code is being touched because if I throw an exception just before calling TrySetCanceled I can see it. So I'm thinking of what other test could be helpful to verify this.